homeassistant/components/esphome/__init__.py

```diff
@@ -653,6 +653,9 @@ async def execute_service(call: ServiceCall) -> None:
 async def _setup_services(
     hass: HomeAssistant, entry_data: RuntimeEntryData, services: list[UserService]
 ) -> None:
+    if entry_data.device_info is None:
+        # Can happen if device has never connected or .storage cleared
+        return
```

Comment:
Will we set up the services later for this device?

Reply:
When the device next connects (and we have received the services data). ESPHome is a bit unique in that it has user-defined services that get transmitted by the device.